plansys2_core/include/plansys2_core/State.hpp

```diff
@@ -0,0 +1,355 @@
+// Copyright 2024 Intelligent Robotics Lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PLANSYS2_CORE__STATE_HPP_
+#define PLANSYS2_CORE__STATE_HPP_
+
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "plansys2_core/DerivedResolutionGraph.hpp"
+#include "plansys2_core/Types.hpp"
+#include "plansys2_msgs/msg/derived.hpp"
+#include "plansys2_msgs/msg/node.hpp"
+#include "plansys2_msgs/msg/param.hpp"
+#include "plansys2_msgs/msg/state.hpp"
+
+namespace plansys2
+{
+
+class State
```

Comment:
I can think of this class, but it would be important to document the class more precisely.